### PR TITLE
Adding AKV support to sqlvm powershell with tests and help files added

### PR DESCRIPTION
**Pull request by tlao33**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@tlao33__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

**Comment by OJDUDE** [comment]
This is the same code change in https://github.com/Azure/azure-powershell/pull/2448 which was already reviewed. Hovsep Mkrtchyan asked Tao to squash into a single commit. Tao created a new PR for this with a single commit.

File: src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml [hunk 1]
```diff
@@ -11738,6 +11738,241 @@ This cmdlet does not generate any output.</maml:para>
 <command:command xmlns:maml="http://schemas.microsoft.com/maml/2004/10" xmlns:command="http://schemas.microsoft.com/maml/dev/command/2004/10" xmlns:dev="http://schemas.microsoft.com/maml/dev/2004/10" xmlns:MSHelp="http://msdn.microsoft.com/mshelp">
 <!--Generated by PS Cmdlet Help Editor-->
 	<command:details>
+		<command:name>New-AzureVMSqlServerKeyVaultCredentialConfig</command:name>
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml]: Please add proper documentation.

Comment (tlao33) [inline src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml]: @hovsepm I have added the document in the help file in the new commit. Please review it. Thank you.

File: src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml [hunk 2]
```diff
@@ -11738,6 +11738,241 @@ This cmdlet does not generate any output.</maml:para>
...
+				<command:parameterValue required="false" variableLength="false">String</command:parameterValue>
+			</command:parameter>
+			<command:parameter required="false" variableLength="false" globbing="false" pipelineInput="false" position="named">
+				<maml:name>ServicePrincipalSecret</maml:name>
+				<maml:description>
+					<maml:para />
+				</maml:description>
+				<command:parameterValue required="false" variableLength="false">SecureString</command:parameterValue>
+			</command:parameter>
+			<command:parameter required="false" variableLength="false" globbing="false" pipelineInput="false" position="named">
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml]: You can remove InformatioNAction and InformationVariable, these are not needed.

File: src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml [hunk 3]
```diff
@@ -11738,6 +11738,241 @@ This cmdlet does not generate any output.</maml:para>
...
+				<maml:para />
+			</maml:description>
+			<command:parameterValue required="false" variableLength="false">SecureString</command:parameterValue>
+			<dev:type>
+				<maml:name>String</maml:name>
+				<maml:uri/>
+			</dev:type>
+			<dev:defaultValue>none</dev:defaultValue>
+		</command:parameter>
+		<command:parameter required="false" variableLength="false" globbing="false" pipelineInput="false" position="named">
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Microsoft.Azure.Commands.Compute.dll-Help.xml]: No need for InfomrationAction/InformationVariable

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/SqlIaaSExtensionTests.cs [hunk 4]
```diff
@@ -15,6 +15,7 @@ public SqlIaaSExtensionTests(Xunit.Abstractions.ITestOutputHelper output)
         public void TestSqlIaaSExtension()
         {
             ComputeTestController.NewInstance.RunPsTest("Test-SetAzureRmVMSqlServerExtension");
+            ComputeTestController.NewInstance.RunPsTest("Test-SetAzureRmVMSqlServerAKVExtension");
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/SqlIaaSExtensionTests.cs]: Same comment as before - please edit the function in the ps1 script to include the new cmdlets, or please add a new test method for thsi new extension.

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+// ----------------------------------------------------------------------------------
+
+using System.Security;
+
+namespace Microsoft.Azure.Commands.Compute
+{
+    /// <summary>
+    /// AKV settings to configure on SQL VM
+    /// </summary>
+    public class PrivateKeyVaultCredentialSettings
```
Comment (markcowl) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs]: .Net coding guidelines:  class name and file name must match.  The class name here is the better of the two.

**Comment by markcowl** [comment]
@tlao33 Gettign much closer.  Still need to properly configure the test as specified - also one .net coding guidelines vioplation to fix and a couple of minor help improvements.  otherwise, this looks good.

**Comment by tlao33** [comment]
@markcowl I did not have the permission to Jekins.

**Comment by markcowl** [comment]
@tlao33 http://aka.ms/azuregithub

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs (hunk 5 above)
Comment (tlao33) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs]: @markcowl I followed the existing code style in this project like file name is AzureVMSqlServerAutoPatchingSettings and class name is AutoPatchingSettings.

Comment (tlao33) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs]: @markcowl I have fixed it

**Comment by tlao33** [comment]
@markcowl The checks failed because of a regression fixed in this PR https://github.com/Azure/azure-powershell/pull/2510/files. Can you please run the tests again

**Comment by tlao33** [comment]
@markcowl  never mind, I found it.

**Comment by markcowl** [comment]
@tlao33 The test is failing (an issue with the mock), and you must respond to or fix review comments.

**Comment by tlao33** [comment]
@markcowl @hovsepm  Please review the new PR of azure-powershell test case.

File: src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs (hunk 5 above)
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute/Extension/SqlServer/AzureVMSqlServerPrivateKeyVaultCredentialSettings.cs]: This is still not fixed.

File: src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/SqlIaaSExtensionTests.ps1 [hunk 6]
```diff
@@ -10,6 +10,115 @@
 	8) Test with incorrect Name and crrect Version
 	9) Test with incorrect Name and incorrect Version
 #>
+
+function Test-SetAzureRmVMSqlServerAKVExtension
+{
+	Set-StrictMode -Version latest; $ErrorActionPreference = 'Stop'
+
+	# Setup
```
Comment (hovsepm) [inline src/ResourceManager/Compute/Commands.Compute.Test/ScenarioTests/SqlIaaSExtensionTests.ps1]: please replace all TAB symbols in this file to 4 Space symbols. That way this file will be properly formatted on github.

**Comment by tlao33** [comment]
@hovsepm  Please review the new commit.

**Comment by markcowl** [comment]
on demand run here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/932/

LGTM once it passes

**Comment by tlao33** [comment]
@markcowl  Please let me know when build complete. Thank you

**Pull request merged**